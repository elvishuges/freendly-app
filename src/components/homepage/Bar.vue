<template>
<v-app-bar
      color="#fcb69f"
      dark
      shrink-on-scroll
      src="https://picsum.photos/1920/1080?random"
    >
      <template v-slot:img="{ props }">
        <v-img
          v-bind="props"
          gradient="to top right, rgba(19,84,122,.5), rgba(128,208,199,.8)"
        ></v-img>
      </template>

      <v-app-bar-nav-icon  @click="ChildtoggleDrawer"></v-app-bar-nav-icon>

      <v-toolbar-title>{{openDrawer}}</v-toolbar-title>

      <v-spacer></v-spacer>

      <v-btn icon>
        <v-icon>mdi-magnify</v-icon>
      </v-btn>

      <v-btn icon>
        <v-icon>mdi-dots-vertical</v-icon>
      </v-btn>
    </v-app-bar>
</template>

<script>
export default {
props: {
    toggleDrawer: { type: Function },
     openDrawer: { type: Boolean },
  },
  data() {
    return {
      drawer:null,
    }
  },
  mounted () {
     this.drawer = this.openDrawer ;
  },

  methods: {
    ChildtoggleDrawer: function () {
           // Do your stuff
           this.toggleDrawer(true);
       }
  },

};
</script>

<style>
</style>