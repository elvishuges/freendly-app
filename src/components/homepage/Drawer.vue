<template>
  <div>
    <v-navigation-drawer
      v-model="showDrawer"
      absolute
      dark
      class="light-blue darken-1"
      height="100%"
      temporary
    >
      <v-list-item class="grey lighten-4">
        <v-list-item-content>
          <v-list-item-title class="title light-blue--text">{{
            infoEmpresa.nome
          }}</v-list-item-title>
          <v-list-item-subtitle class="light-blue--text">{{
            infoEmpresa.descricao
          }}</v-list-item-subtitle>
        </v-list-item-content>
      </v-list-item>
      <v-divider></v-divider>
      <v-list nav>
        <v-list-item
          v-for="item in items"
          :key="item.title"
          link
          :to="{ path: '/' + item.route }"
        >
          <v-list-item-icon>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>{{ item.title }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
        <v-list-item @click="logout()">
          <v-list-item-icon>
            <v-icon>mdi-logout</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>Logout</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>

    <!-- * bar superior * -->

    <v-app-bar
      color="#fcb69f"
      dark
      src="https://picsum.photos/1920/1080?random"
    >
      <template v-slot:img="{ props }">
        <v-img
          v-bind="props"
          gradient="to top right, rgba(19,84,122,.5), rgba(128,208,199,.8)"
        ></v-img>
      </template>

      <v-app-bar-nav-icon
        @click="showDrawer = !showDrawer"
      ></v-app-bar-nav-icon>

      <v-toolbar-title>showDrawer</v-toolbar-title>

      <v-spacer></v-spacer>

      <v-btn icon>
        <v-icon>mdi-magnify</v-icon>
      </v-btn>

      <v-btn icon>
        <v-icon>mdi-dots-vertical</v-icon>
      </v-btn>
    </v-app-bar>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import { AUTH_LOGOUT } from "./../../store/actions/auth";
import { CLEAN_COMPANY_DATE } from "./../../store/actions/user";

export default {
  props: {
    openDrawer: { type: Boolean },
  },
  data() {
    return {
      key: "value",
      showDrawer: false,
      items: [
        {
          title: "Dashboard",
          icon: "mdi-puzzle-edit-outline",
          route: "dashboard",
        },
        {
          title: "Projeto",
          icon: "mdi-plus",
          route: "formProjeto",
        },
        {
          title: "Jobs",
          icon: "mdi-eye",
          route: "searchJobs",
        },
        {
          title: "Editar dados",
          icon: "mdi-content-save-edit",
          route: "",
        },
      ],
    };
  },
  mounted() {},
  methods: {
    logout() {
      console.log("logout");
      this.$store.dispatch(AUTH_LOGOUT).then(() => {
        this.$store.dispatch(CLEAN_COMPANY_DATE);
        this.$router.push("/login");
      });
    },
  },

  computed: {
    ...mapGetters([
      "infoEmpresa",
      // ...
    ]),
  },
};
</script>

<style>
</style>