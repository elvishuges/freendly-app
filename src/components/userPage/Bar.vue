<template>
  <v-app-bar color="#fcb69f" dark src="https://picsum.photos/1920/1080?random">
    <template v-slot:img="{ props }">
      <v-img
        v-bind="props"
        gradient="to top right, rgba(19,84,122,.5), rgba(128,208,199,.8)"
      ></v-img>
    </template>

    <v-app-bar-nav-icon @click="openDrawer"></v-app-bar-nav-icon>

    <v-toolbar-title>{{ title }}</v-toolbar-title>
  </v-app-bar>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: "TÃ­tulo",
    },
  },
  data() {
    return {
      drawer: null,
    };
  },
  mounted() {
    this.drawer = this.openDrawer;
  },

  methods: {
    openDrawer: function () {
      this.$emit("open-drawer");
    },
  },
};
</script>

<style>
</style>