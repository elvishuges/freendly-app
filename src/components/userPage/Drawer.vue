<template>
  <div>
    <v-navigation-drawer
      v-bind:value="drawer"
      v-on:input="$emit('updateDrawer', $event)"
      absolute
      dark
      class="light-blue darken-1"
      height="100%"
      temporary
    >
      <v-list-item class="grey lighten-4">
        <v-list-item-content>
          <v-list-item-title class="title light-blue--text">{{
            infoEmpresa.nome
          }}</v-list-item-title>
          <v-list-item-subtitle class="light-blue--text">{{
            infoEmpresa.descricao
          }}</v-list-item-subtitle>
        </v-list-item-content>
      </v-list-item>
      <v-divider></v-divider>
      <v-list nav>
        <v-list-item
          v-for="item in items"
          :key="item.title"
          link
          :to="{ path: item.route }"
        >
          <v-list-item-icon>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>{{ item.title }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
        <v-list-item @click="logout()">
          <v-list-item-icon>
            <v-icon>mdi-logout</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>Logout</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import { EventBus } from "./../../utils/event-bus.js";
import { AUTH_LOGOUT } from "./../../store/actions/auth";
import { CLEAN_COMPANY_DATE } from "./../../store/actions/user";

export default {
  props: {
    drawer: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      key: "value",
      items: [
        {
          title: "Dashboard",
          icon: "mdi-puzzle-edit-outline",
          route: "/dashboard",
        },
        {
          title: "Projeto",
          icon: "mdi-plus",
          route: "/projeto/add",
        },
        {
          title: "Jobs",
          icon: "mdi-eye",
          route: "/projetos",
        },
        {
          title: "Editar dados",
          icon: "mdi-content-save-edit",
          route: "/#",
        },
      ],
    };
  },
  mounted() {},
  methods: {
    logout() {
      console.log("logout");
      this.$store.dispatch(AUTH_LOGOUT).then(() => {
        this.$router.push("/login");
        this.$store.dispatch(CLEAN_COMPANY_DATE);
      });
    },
    handleClickOutside() {
      console.log("aquiii");
    },

    GlobalEventshowChatDrawer() {
      // Send the event on a channel (i-got-clicked) with a payload (the click count.)
      EventBus.$emit("GlobalEventshowChatDrawer", true);
    },
  },

  computed: {
    ...mapGetters([
      "infoEmpresa",
      // ...
    ]),
  },
};
</script>

<style>
</style>