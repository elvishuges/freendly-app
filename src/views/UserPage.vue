<template>
  <v-app>
    <drawer
      :drawerItens="drawerItens"
      @updateDrawer="updateDrawer"
      :drawer="drawer"
    />
    <bar title="UsuÃ¡rio" @open-drawer="handleOpenDrawerEvent" />

    <router-view></router-view>
  </v-app>
</template>

<script>
import Drawer from "./../components/userPage/Drawer";
import Bar from "./../components/userPage/Bar";

export default {
  components: {
    Drawer,
    Bar,
  },
  data() {
    return {
      drawer: false,
      key: "value",
      items: [
        { title: "Home", icon: "dashboard" },
        { title: "About", icon: "question_answer" },
      ],
      drawerItens: [
        {
          title: "Dashboard",
          icon: "mdi-puzzle-edit-outline",
          route: "/dashboard",
        },
        {
          title: "Projeto",
          icon: "mdi-plus",
          route: "/projeto/add",
        },
        {
          title: "Jobs",
          icon: "mdi-eye",
          route: "/projetos",
        },
        {
          title: "Editar dados",
          icon: "mdi-content-save-edit",
          route: "/#",
        },
      ],
    };
  },

  mounted() {},

  methods: {
    handleClickOutside() {
      console.log("log", this.drawer);
      this.drawer ? (this.drawer = false) : "";
    },
    handleOpenDrawerEvent() {
      this.drawer = true;
    },
    updateDrawer(payload) {
      payload ? "" : (this.drawer = false);
    },
  },
};
</script>

<style>
</style>