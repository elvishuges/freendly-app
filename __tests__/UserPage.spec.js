

//Utilities
import VueRouter from 'vue-router';
import Vuetify from 'vuetify';


// Components
import UserPage from '@/views/UserPage'
import Drawer from '@/components/userPage/Drawer'
import Bar from '@/components/userPage/Bar'

// Utilities
import store from "@/store";
import {
    createLocalVue,
    mount,
    shallowMount
} from '@vue/test-utils'

const localVue = createLocalVue()

describe('Userpage Tests', () => {
    const build = () => {
        const wrapper = shallowMount(UserPage, {
            localVue: localVue,
            store,
            vuetify,
        })
        return {
            wrapper,
            drawer: () => wrapper.findComponent(Drawer),
            bar: () => wrapper.findComponent(Bar),

        }
    }

    let vuetify

    beforeEach(() => {
        jest.resetAllMocks()
        const localVue = createLocalVue();
        localVue.use(VueRouter);
        vuetify = new Vuetify()
    });

    it('create components and snapshots', () => {
        const { wrapper } = build()
        expect(wrapper.html()).toMatchSnapshot()
    })

    it("render the two children componens", () => {
        const { wrapper } = build()
        expect(wrapper.findComponent(Drawer).exists()).toBe(true)
        expect(wrapper.findComponent(Bar).exists()).toBe(true)
    })

    it("render drawer and bar", () => {
        const { drawer, bar } = build()
        expect(drawer).toBeTruthy();
        expect(bar).toBeTruthy();
    })

    it("bar event to open drawer", async () => {
        const { drawer, wrapper, bar } = build()
        bar().vm.$emit('open-drawer')
        await bar().vm.$nextTick() // Wait until $emits have been handled
        expect(drawer().vm.drawer).toBe(true)
    })

})