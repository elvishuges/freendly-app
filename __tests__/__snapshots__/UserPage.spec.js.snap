// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Userpage Tests create components and snapshots 1`] = `
<v-app-stub id="app">
  <drawer-stub draweritens="[object Object],[object Object],[object Object],[object Object]"></drawer-stub>
  <bar-stub title="Usuário"></bar-stub>
  <router-view></router-view>
</v-app-stub>
`;

exports[`Userpage Tests criacao de component 1`] = `
<v-app-stub id="app">
  <drawer-stub draweritens="[object Object],[object Object],[object Object],[object Object]"></drawer-stub>
  <bar-stub title="Usuário"></bar-stub>
  <router-view></router-view>
</v-app-stub>
`;
